app/src/main/java/com/jrod7938/textchangeapp/screens/login/LoginScreenViewModel.kt

```diff
@@ -89,13 +89,31 @@ class LoginScreenViewModel: ViewModel() {
      */
     fun signInWithEmailAndPassword(email: String, password: String, home: () -> Unit) = viewModelScope.launch{
         try {
+            _loading.value = true
             auth.signInWithEmailAndPassword(email, password)
                 .addOnCompleteListener{ task ->
                     if (task.isSuccessful){
                         Log.d("Firebase", "signInWithEmailAndPassword: ${task.result}")
-                        home()
+                        FirebaseFirestore.getInstance().collection("users")

```

Comment:
this is unnecessary, the task of logging the user in has already been complete therefore, all you have to do is navigate to the Home Screen. No need to make extra firebase calls

Reply:
I fixed this in #948b1f4

Reply:
Please read my PR. There is a bug. A user can bypass verification by closing the app. When they reopen it, they log in with their credentials and have access. The app however, loads incorrectly because the user has no information, just an email and a password. This is bc the rest of the user's full details is written into the database after successful verification, but in this case verification hasn't been completed. To avoid this, I check to see if the user attempting to log in has a full user profile in the database. In this case, I check if they have a display name. If they do, log in.

Reply:
Here is a video of what happens in this case: https://drive.google.com/file/d/1GLyQPUSfpMKnI9GvbGUnRMx1MmWL8PEE/view?usp=sharing

Reply:
This is what happens when the unauthorized user tries to actually use the app: https://drive.google.com/file/d/1GUJICDJ_l8ESsmWCy-mmwmY8nOZIhz06/view?usp=sharing

Reply:
issue is fixed in 25d17aa

Reply:
No this doesn't work. In this case, it's not that ther verification fails. It's bypassed altogether. So it never reaches this step and the user is not deleted.

Reply:
I don't think this issue can be fixed then. The delete user function runs on the users device so if they cancel the app it never runs.

Reply:
Yes. I did fix it. It probably isn't the best implementation, but it did work. Thats why I did that second data base check.

Reply:
I see, I pushed your update again

Reply:
splash screen now checks if the Email is verified, if it is go home else go to login screen